ndcube/ndcube.py

```diff
@@ -399,6 +400,10 @@ def __init__(self, data, wcs=None, uncertainty=None, mask=None, meta=None,
                 global_coords = deepcopy(global_coords)
             self._global_coords = global_coords
 
+        # If meta is axis-aware, make it to have same shape as cube.
+        if isinstance(self.meta, NDMetaABC):
+            self.meta.data_shape = self.shape
```

Comment:
Should this be in the setter for `.meta`? Do we have a setter for `.meta`?

Reply:
We don't.  We outsource that to `NDData`.